typescript/src/redemption.ts

```diff
@@ -406,3 +409,79 @@ export async function getRedemptionRequest(
 
   return redemptionRequests[0]
 }
+
+/**
+ * Finds the oldest active wallet that has enough BTC to handle a redemption request.
+ * @param amount The amount to be redeemed.
+ * @param bridge The handle to the Bridge on-chain contract.
+ * @param bitcoinClient Bitcoin client used to interact with the network.
+ * @param bitcoinNetwork Bitcoin network.
+ * @returns Promise with the wallet details needed to request a redemption.
+ */
+export async function findWalletForRedemption(
+  amount: BigNumber,
+  bridge: Bridge,
+  bitcoinClient: BitcoinClient,
+  bitcoinNetwork: BitcoinNetwork
```

Comment:
Instead of passing the `bitcoinNetwork` we could pull it from `bitcoinClient.getNetwork()`. But this would require an additional call to the Electrum server, so I'm not sure if it's worth it.

Reply:
I'm ok with both approaches. Decided to pass the `bitcoinNetwork` because we pass it as well in `calculateDepositAddress` function so we are consistent. But on the other hand the `calculateDepositAddress` does not accept the `BitcoinClient` so there is no other way to get the Bitcoin network 🤔.